src/util/params.js

```diff
@@ -24,7 +24,10 @@ export function fillParams (
 
     return filler(params, { pretty: true })
   } catch (e) {
-    if (process.env.NODE_ENV !== 'production') {
+    if (
+      process.env.NODE_ENV !== 'production' &&
+      params.pathMatch !== '' // Fix #3072 no warn if `pathMatch` is empty
```

Comment:
let's check if typeof is not string on the warn method (replacing the `false` value)

Reply:
fixed at https://github.com/vuejs/vue-router/pull/3081/commits/590adbbf1e78662e58fc26ee656ea600261db1c4.
I also simplify source code at https://github.com/vuejs/vue-router/pull/3081/commits/8a213d9f49f0f22b858672879b5077130a3a3db4.